src/number/format-decimal.js

```diff
@@ -42,7 +51,11 @@ module.exports = function(value, localeData, options) {
 
 		digitNum++;
 
-		if (digitNum > 1 && digitNum % localeData.groupSize === 1) {
+		if (digitNum > 1 && digitNum % currentGroupSize === 1) {
+			if ( groupSizes.length > 0) {
+				currentGroupSize = groupSizes.shift();
+			}
+			digitNum = 1;	// reset this back to 1 to handle variable group sizes
```

Comment:
Perhaps? The word _reset_ here is what made me notice, and I would assume it was a bug if I saw it ("oops, they accidently 'reset' it to the wrong value"). So yeah, either needs to be explained differently, or the arrangement should change. What happens when the initial value of `digitNum` is 1 instead of `0`?